core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -55,77 +55,139 @@ object ConsumerGroupCommand {
+        println("Note: This will only show information about consumers that use the Java consumer API (non-ZooKeeper-based consumers).\n")
         new KafkaConsumerGroupService(opts)
       }
     }
 
     try {
       if (opts.options.has(opts.listOpt))
-        consumerGroupService.list()
-      else if (opts.options.has(opts.describeOpt))
-        consumerGroupService.describe()
+        consumerGroupService.listGroups().foreach(println(_))
+      else if (opts.options.has(opts.describeOpt)) {
+        val (state, assignments) = consumerGroupService.describeGroup()
+        val groupId = opts.options.valuesOf(opts.groupOpt).asScala.head
+        assignments match {
+          case None =>
+            printError(s"The consumer group '$groupId' does not exist.")
+          case Some(assignments) =>
+            if (assignments.isEmpty)
+              state match {
+                case Some("Dead") =>
+                  printError(s"Consumer group '$groupId' does not exist.")
+                case Some("Empty") =>
+                  printError(s"Consumer group '$groupId' has no active members.")
+                case Some(_) =>
+                  printError(s"Consumer group '$groupId' is rebalancing.")
+                case None =>
+                  // the control should never reach here
+                  throw new KafkaException("Expected a valid consumer group state, but none found.")
+              }
+            else
+              printAssignment(assignments)
+        }
+      }
       else if (opts.options.has(opts.deleteOpt)) {
         consumerGroupService match {
-          case service: ZkConsumerGroupService => service.delete()
-          case _ => throw new IllegalStateException(s"delete is not supported for $consumerGroupService")
+          case service: ZkConsumerGroupService => service.deleteGroups()
+          case _ => throw new IllegalStateException(s"delete is not supported for $consumerGroupService.")
         }
       }
     } catch {
       case e: Throwable =>
-        println("Error while executing consumer group command " + e.getMessage)
-        println(Utils.stackTrace(e))
+        printError(s"Executing consumer group command failed due to ${e.getMessage}", Some(e))
     } finally {
       consumerGroupService.close()
     }
   }
 
+  def printError(msg: String, e: Option[Throwable] = None): Unit = {
+    println(s"Error: $msg")
+    e.foreach(debug("Exception in consumer group command", _))
+  }
+
+  def printAssignment(groupAssignment: Seq[PartitionAssignmentState]): Unit = {
+    require(!groupAssignment.isEmpty)
+    println("Consumer group: " + groupAssignment.head.group)
+    groupAssignment.head.coordinator match {
+      case Some(node) => println(s"Consumer group coordinator broker id: ${node.id}")
+      case default =>
+    }
+    println("\n%-30s %-10s %-15s %-15s %-10s %-50s %-30s %s".format("TOPIC", "PARTITION", "CURRENT-OFFSET", "LOG-END-OFFSET", "LAG", "MEMBER-ID", "MEMBER-HOST", "CLIENT-ID"))
```

Comment:
Couple naming suggestions:

1. `MEMBER-ID` -> `CONSUMER-ID`
2. `MEMBER-HOST` -> `HOST`

What do you think? Also, it seems neither of these options are available for the old consumer. Maybe it would reduce the noise if we leave them out of the output in that case? Not sure if there's a clean way to do that though.